Cython/Utility/CppConvert.pyx

```diff
@@ -52,14 +52,24 @@ cdef vector[X] {{cname}}(object o) except *:
 #################### vector.to_py ####################
 
 cdef extern from *:
-    cdef cppclass vector "const std::vector" [T]:
+    cdef cppclass vector "std::vector" [T]:
         size_t size()
         T& operator[](size_t)
 
-@cname("{{cname}}")
-cdef object {{cname}}(vector[X]& v):
-    return [v[i] for i in range(v.size())]
+cdef extern from "Python.h":
+    void Py_INCREF(object)
+    list PyList_New(Py_ssize_t size)
+    void PyList_SET_ITEM(object list, Py_ssize_t i, object o)
 
+@cname("{{cname}}")
+cdef object {{cname}}(const vector[X]& v):
+    o = PyList_New(v.size())
+    cdef size_t i = 0
```

Comment:
Does this need to be `size_t` instead of the usual `Py_ssize_t` for indexing?
```suggestion
    cdef Py_ssize_t i
```

Reply:
This might just be a mismatch where no option can ever be "right" - `size_t` matches the type of of `v.size()`, but not `PyList_SET_ITEM`.

Possibly the right thing to do is to raise an error if `v.size() > PY_SSIZE_T_MAX` and then use `Py_ssize_t` after that check? But I doubt if many people have the memory to run that test-case.

Reply:
I changed it to Py_ssize_t. Even for a bit vector (vector<bool>) this would require 1.07E+09 Gb of memory, so I do not think, that this is a concern (+ the memory allocation for the PyList). In Cpython this would already causes a exception because a negative size is specified, while in PyPy this would allocate a empty list.
https://github.com/mozillazg/pypy/blob/11af013b3723c15c9ce51f68fd5c2db51f74ba3d/pypy/module/cpyext/listobject.py#L23
I personally do not see anyone running into this issue, given the big memory requirement